src/logics/syncswap/logic.swap-token.test.ts

```diff
@@ -0,0 +1,119 @@
+import { LogicTestCase } from 'test/types';
+import { Router__factory } from './contracts';
+import { SwapTokenLogic, SwapTokenLogicFields, SwapTokenLogicOptions } from './logic.swap-token';
+import * as common from '@protocolink/common';
+import { constants, utils } from 'ethers';
+import * as core from '@protocolink/core';
+import { expect } from 'chai';
+import { getContractAddress } from './configs';
+import { zksyncTokens } from './tokens';
+
+describe('SyncSwap SwapTokenLogic', function () {
+  context('Test getTokenList', async function () {
+    SwapTokenLogic.supportedChainIds.forEach((chainId) => {
+      it(`network: ${common.toNetworkId(chainId)}`, async function () {
+        const logic = new SwapTokenLogic(chainId);
+        const tokenList = await logic.getTokenList();
+        expect(tokenList).to.have.lengthOf.above(0);
+      });
+    });
+  });
+
+  context('Test build', function () {
+    const chainId = common.ChainId.zksync;
+    const logic = new SwapTokenLogic(chainId);
+    const routerAddress = getContractAddress(chainId, 'Router');
+    const iface = Router__factory.createInterface();
+
+    const testCases: LogicTestCase<SwapTokenLogicFields, SwapTokenLogicOptions>[] = [
+      {
+        fields: {
+          input: new common.TokenAmount(zksyncTokens.ETH, '1'),
+          output: new common.TokenAmount(zksyncTokens.USDC, '0'),
+          paths: [
+            {
+              steps: [
+                {
+                  pool: '0x80115c708E12eDd42E504c1cD52Aea96C547c05c',
+                  tokenIn: '0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91',
```

Comment:
Is it a good idea to use constants to represent these addresses?

Reply:
[Oral Discussion]